src/jira.py

```diff
@@ -99,59 +115,82 @@ def get_github_equivalent(self):
 
         # Issue status has to be updated as a transition
-        r = requests.post(f"{self.url}issue/{self.jira_key}/transitions", headers=self.headers, json=transition)
+        r = requests.post(f'{self.url}issue/{self.jira_key}/transitions', headers=self.headers, json=transition)
 
         if r.status_code != 204:  # HTTP 204 No Content on success
-            print(f"{r.status_code} Error transitioning")
+            print(f'{r.status_code} Error transitioning')
 
         # Issue assignee, description, summary, and story points fields can be updated from a dictionary
-        r = requests.put(f"{self.url}issue/{self.jira_key}", headers=self.headers, json=self.dict_format())
+        r = requests.put(f'{self.url}issue/{self.jira_key}', headers=self.headers, json=self.dict_format())
 
         if r.status_code != 204:  # HTTP 204 No Content on success
-            print(f"{r.status_code} Error")
+            print(f'{r.status_code} Error: {r.reason}')
 
     def post_new_issue(self):
         """Post this issue to Jira for the first time. The issue must not already exist."""
 
-        r = requests.post(self.url + "issue/", headers=self.headers, json=self.dict_format())
+        r = requests.post(f'{self.url}issue/', headers=self.headers, json=self.dict_format())
 
         if r.status_code != 201:  # HTTP 201 means created
-            print(f"{r.status_code} Error")
+            print(f'{r.status_code} Error: {r.reason}')
 
-        self.jira_key = r.json()["key"]  # keep the key that Jira assigned to this issue when creating it
-
-    def add_to_epic(self, epic_key):
-        """Make this issue part of a Jira epic. If it is already in an epic, that will be overwritten."""
-        issues = {'issues': [self.jira_key]}
+        self.jira_key = r.json()['key']  # keep the key that Jira assigned to this issue when creating it
 
+    def add_to_this_epic(self, issue_key):
+        """Make the given issue belong to this epic (self). If it is already in an epic, that will be overwritten."""
+        issues = {'issues': [issue_key]}
+        old_api_url = first(self.url.split('api')) # remove 'api/latest' from the url
         # This operation seems to work only in the old API version 1.0
-        r = requests.post(f"https://ucsc-cgl.atlassian.net/rest/agile/1.0/epic/{epic_key}/issue", json=issues,
-                          headers=self.headers)
+        r = requests.post(f'{old_api_url}agile/1.0/epic/{self.jira_key}/issue', json=issues, headers=self.headers)
 
         if r.status_code != 204:  # HTTP 204 No content on success
-            print(f"{r.status_code} Error")
+            print(f'{r.status_code} Error: {r.reason}')
+
+    def remove_from_this_epic(self, issue_key):
+        issues = {'issues': [issue_key]}
+        old_api_url = first(self.url.split('api'))  # remove 'api/latest' from the url
+        # TODO is it a problem that this functionality only exists in the old api version
```

Comment:
It's good that you point this out. We could make a request in the Jira repo to ask if they can add this functionality to the new version, or whether they covered it elsewhere now.

Reply:
Sent a question about this to atlassian support